src/l500/l500-color.cpp

```diff
@@ -221,7 +230,70 @@ namespace librealsense
                 return intrinsics;
             }
         }
-        throw std::runtime_error( to_string() << "intrinsics for resolution " << profile.width << "," << profile.height << " don't exist" );
+        throw std::runtime_error( to_string() << "intrinsics for resolution " << width << ","
+                                              << height << " don't exist" );
+    }
+
+    double l500_color_sensor::read_temperature() const
+    {
+        auto hwm = *_owner->_hw_monitor;
+
+        std::vector< byte > res;
+
+        try
+        {
+            res = hwm.send( command{ TEMPERATURES_GET } );
+        }
+        catch( std::exception const & e )
+        {
+            LOG_ERROR(
+                "Failed to get temperatures; hardware monitor in inaccessible: " << e.what() );
+            return 0.;
+        }
+
+        if( res.size() < sizeof( temperatures ) )  // New temperatures may get added by FW...
+        {
+            LOG_ERROR( "Failed to get temperatures; result size= "
+                       << res.size() << "; expecting at least " << sizeof( temperatures ) );
+            return 0.;
+        }
+        auto const & ts = *( reinterpret_cast< temperatures * >( res.data() ) );
+        LOG_DEBUG( "HUM temperture is currently " << ts.HUM_temperature << " degrees Celsius" );
```

Comment:
This was AC_LOG before

Reply:
This code moved to l500-color.cpp and can be used outside AC.